ddp-workspace/projects/ddp-pancan/src/app/components/splash-pages/lms-page/lms-page.component.ts

```diff
@@ -11,19 +17,41 @@ import { FormControl } from '@angular/forms';
 export class LmsPageComponent implements OnInit {
     readonly phone: string;
     readonly email: string;
-    emailToNotify: FormControl;
     readonly AppRoutes = AppRoutes;
 
-    constructor(@Inject('toolkit.toolkitConfig') config: ToolkitConfigurationService) {
+    constructor(
+        private dialog: MatDialog,
+        @Inject('toolkit.toolkitConfig') private config: ToolkitConfigurationService,
+        private translateService: TranslateService,
+        private log: LoggingService
+    ) {
         this.phone = config.lmsPagePhone;
         this.email = config.lmsPageEmail;
     }
 
+    notify(): void {
+        this.dialog.open(JoinMailingListComponent, {
+            ...JOIN_MAILING_LIST_DIALOG_SETTINGS,
+            data: {
+                studyGuid: this.config.lmsStudyGuid,
+                useLanguage: 'en'
```

Comment:
does it mean that we should always open JoinMailingList dialog in english only for all languages? if not, could you please explain this part a liitle

Reply:
We got a message from CMI that LMs page will not be translated. 
It means we should keep the page (and related stuff like joinMailingList) in English even though a user can switch to another language (Spanish) for whole study.
Other pages and related joinMailingList dialog should follow chosen language.

Reply:
2 questions then: should we switch back to the selected language when we navigate from lms page and isn't enough to switch englush translation only in lms parent component? why do we need to to it one more time in dialog component? the service is singletone and should be global for the whole app

Reply:
> should we switch back to the selected language when we navigate from lms page

We should not. We do not change the current language in translate service
![q1](https://user-images.githubusercontent.com/7396837/129067910-4ff8d6b5-422a-447c-b6c0-7c3e4e7a9aeb.png)

Reply:
>isn't enough to switch englush translation only in lms parent component? why do we need to to it one more time in dialog component?

It is not enough. I believe it is because we use `JoinMailingList.component` not directly on LMS page , but as an entry component.